utils/helpers/getLocaleYear.ts

```diff
@@ -1,19 +1,22 @@
-// Imports
 import { LangCode } from "@/utils/types/common";
 
+const BE_AD_DIFFERENCE = 543;
+
+/**
+ * Converts a year between the Buddhist Era (BE) and the Anno Domini (AD) calendar.
+ *
+ * @param locale The locale to convert the year to.
+ * @param year The year to convert.
+ * @param fromType The type of the year to convert from, either "AD" or "BE".
+ *
+ * @returns The converted year.
+ */
 export default function getLocaleYear(
   locale: LangCode,
   year: number,
-  fromType?: "AD" | "BE",
+  fromType: "AD" | "BE" = "AD",
 ): number {
-  return (
-    year +
-    // From BE to AD (year - 543)
-    (fromType === "BE" && locale === "en-US"
-      ? -543
-      : // From AD to BE (year + 543)
-      fromType === "AD" && locale === "th"
-      ? 543
-      : 0)
-  );
+  if (fromType === "BE" && locale === "en-US") return year - BE_AD_DIFFERENCE;
```

Comment:
The conversion is much cleaner and more practical to use in further development.